File: src/main/java/com/prography/zone_2_be/global/entity/BaseEntity.java
package com.prography.zone_2_be.global.entity;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.prography.zone_2_be.global.converter.InstantToEpochSecondConverter;

import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;

@Getter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@CreatedDate
	@Column(nullable = false, updatable = false, columnDefinition = "BIGINT")
	@Convert(converter = InstantToEpochSecondConverter.class)
	private Long createdAt;

}

File: src/main/kotlin/com/prography/zone_2_be/global/entity/BaseEntity.kt
package com.prography.zone_2_be.global.entity

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.Instant

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseEntity {

    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdAt: Long = Instant.now().toEpochMilli()

    @LastModifiedDate
    @Column(nullable = false)
    var updateAt: Long = Instant.now().toEpochMilli()
}